storage/ndb/rest-server/data-access-rondb/src/rdrs_rondb_connection.cpp

```diff
@@ -52,8 +54,9 @@ RS_Status RDRSRonDBConnection::Init(const char *connection_string, Uint32 connec
   __instance->stats.is_shutdown                 = false;
   __instance->stats.connection_state            = DISCONNECTED;
 
-  __instance->connection_string = reinterpret_cast<char *>(malloc(strlen(connection_string)));
-  std::strncpy(__instance->connection_string, connection_string, strlen(connection_string));
+  size_t connectLength = strnlen(connection_string, 1024) + 1;
```

Comment:
@smkniazi, most importantly, replaced `strlen` with `strnlen` + a max value. Also added +1 for strlen to include the NULL terminata.